src/realm/sync/noinst/client_impl_base.cpp

```diff
@@ -2304,13 +2335,24 @@ Status Session::receive_ident_message(SaltedFileIdent client_file_ident)
     bool legal_at_this_time = (m_bind_message_sent && !have_client_file_ident() && !m_error_message_received &&
                                !m_unbound_message_received);
     if (REALM_UNLIKELY(!legal_at_this_time)) {
-        return {ErrorCodes::SyncProtocolInvariantFailed, "Received IDENT message when it was not legal"};
+        std::string_view illegal_reason;
+        if (m_bind_message_sent)
+            illegal_reason = "BIND message not sent";
+        else if (have_client_file_ident())
+            illegal_reason = "file ident has already been assigned to session";
+        else if (m_error_message_received)
+            illegal_reason = "error message has been received from the server";
+        else if (m_unbound_message_received)
+            illegal_reason = "session has already been suspended";
+        return {ErrorCodes::SyncProtocolInvariantFailed,
```

Comment:
did something happen during testing that required all these extra details? I think even with these extra messages we'd need a full log file to really see what's going on, and I don't know what the user is supposed to do with "BIND message not sent"

Reply:
I agree - I just reverted these changes.